rtt_ros2_params/src/rtt_ros2_params.cpp

```diff
@@ -0,0 +1,296 @@
+}
+
+rclcpp::ParameterValue Params::getParameter(const std::string param_name)
+{
+  RTT::Logger::In in(getName());
+
+  RTT::log(RTT::Debug) <<
+    "Retrieving the parameter \"" << param_name << "\"" <<
+    RTT::endlog();
+
+  rclcpp::ParameterValue paramvalue;
+
+  rclcpp::Node::SharedPtr rosnode;
+  if (get_ros2_node(rosnode)) {
+    rosnode->get_parameter(param_name, paramvalue);
+  } else {
+    RTT::log(RTT::Error) <<
+      "The ROS2 node doesn't exist, no parameter can be retrieved. Import rtt_ros2_node first." <<
+      RTT::endlog();
+  }
+
+  return paramvalue;
+}
+
+bool Params::setParameter(const std::string param_name, const rclcpp::ParameterValue paramvalue)
+{
+  RTT::Logger::In in(getName());
+
+  RTT::log(RTT::Debug) <<
+    "Setting the parameter \"" << param_name << "\"" <<
+    RTT::endlog();
+
+  rclcpp::Node::SharedPtr rosnode;
+  if (get_ros2_node(rosnode)) {
+    if (!rosnode->has_parameter(param_name)) {
+      // Create new parameter
+      RTT::log(RTT::Warning) <<
+        "The parameter did not exist, creating one" <<
+        RTT::endlog();
+      rosnode->declare_parameter(param_name);
+    }
+
+    // Update the parameter
+    try {
+      const auto result = rosnode->set_parameter(rclcpp::Parameter(param_name, paramvalue));
+      if (!result.successful) {
+        RTT::log(RTT::Error) <<
+          "The parameter " << param_name <<
+          " couldn't be set with reason: " << result.reason << RTT::endlog();
+        return false;
+      }
+    } catch (std::exception & e) {
+      RTT::log(RTT::Error) <<
+        "The parameter failed to be set, reason: " <<
+        e.what() << RTT::endlog();
+    }
+  } else {
+    RTT::log(RTT::Error) <<
+      "The ROS2 node doesn't exist, no parameter can be set. Import rtt_ros2_node first and use "
+      "the \"ros2-node\" service." <<
+      RTT::endlog();
+    return false;
+  }
+  return true;
+}
+
+bool Params::loadProperty(const std::string param_name, const std::string property_name)
+{
+  RTT::Logger::In in(getName());
+
+  rclcpp::ParameterValue paramvalue = getParameter(param_name);
+  if (rclcpp::PARAMETER_NOT_SET == paramvalue.get_type()) {
+    RTT::log(RTT::Error) <<
+      "The parameter \"" << param_name << "\" couldn't be retrieved" <<
+      RTT::endlog();
+    return false;
+  }
+
+  if (nullptr != getOwner()->properties()->getProperty(property_name)) {
+    RTT::log(RTT::Debug) <<
+      getOwner()->provides()->getName() <<
+      " has a property " << property_name << RTT::endlog();
+  } else {
+    RTT::log(RTT::Info) <<
+      getOwner()->provides()->getName() <<
+      " has NO property " << property_name << RTT::endlog();
+    for (const auto property : getOwner()->properties()->getProperties() ) {
+      RTT::log(RTT::Debug) <<
+        "Available property: " << property->getName() <<
+        RTT::endlog();
+    }
+  }
+
+  auto prop = getOwner()->properties()->getProperty(property_name);
```

Comment:
Move to the top, such that `getOwner()->properties()->getProperty(property_name)` only needs to be called once in this method.